### PR TITLE
Add docker seliux policy for rpm

### LINKED ISSUE #15498: 1.8.0: Systemd can't start docker on Centos 7.1
Using the default configuration of the latest docker 1.8.0 release, systemd can't start docker.
**Workaround:** `yum install docker-selinux`

Use the commands below to provide key information from your environment:

docker version: 1.8.0-1
docker info: N/A
uname -a: Linux cent7 3.10.0-229.7.2.el7.x86_64 #1 SMP Tue Jun 23 22:06:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Provide additional environment details (AWS, VirtualBox, physical, etc.):

Reproduction:

```
[root@cent7 ~]# uname -a
Linux cent7 3.10.0-229.7.2.el7.x86_64 #1 SMP Tue Jun 23 22:06:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
[root@cent7 ~]# rpm -qa | grep docker
[root@cent7 ~]# yum install -y http://yum.dockerproject.org/repo/main/centos/7/Packages/docker-engine-1.8.0-1.el7.centos.x86_64.rpm
Loaded plugins: fastestmirror
docker-engine-1.8.0-1.el7.centos.x86_64.rpm                                                                                            | 6.2 MB  00:00:03     
Examining /var/tmp/yum-root-ctX5iw/docker-engine-1.8.0-1.el7.centos.x86_64.rpm: docker-engine-1.8.0-1.el7.centos.x86_64
Marking /var/tmp/yum-root-ctX5iw/docker-engine-1.8.0-1.el7.centos.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.8.0-1.el7.centos will be installed
--> Processing Dependency: libcgroup for package: docker-engine-1.8.0-1.el7.centos.x86_64
Loading mirror speeds from cached hostfile
 * base: mirror.evolvedservers.com
 * extras: centos.mbni.med.umich.edu
 * updates: mirror.cc.columbia.edu
--> Running transaction check
---> Package libcgroup.x86_64 0:0.41-8.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================================================================================
 Package                        Arch                    Version                               Repository                                                 Size
==============================================================================================================================================================
Installing:
 docker-engine                  x86_64                  1.8.0-1.el7.centos                    /docker-engine-1.8.0-1.el7.centos.x86_64                   27 M
Installing for dependencies:
 libcgroup                      x86_64                  0.41-8.el7                            base                                                       64 k

Transaction Summary
==============================================================================================================================================================
Install  1 Package (+1 Dependent package)

Total size: 27 M
Total download size: 64 k
Installed size: 27 M
Downloading packages:
libcgroup-0.41-8.el7.x86_64.rpm                                                                                                    |  64 kB  00:00:15     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : libcgroup-0.41-8.el7.x86_64                                                                                                            1/2 
  Installing : docker-engine-1.8.0-1.el7.centos.x86_64                                                                                                2/2 
  Verifying  : docker-engine-1.8.0-1.el7.centos.x86_64                                                                                                1/2 
  Verifying  : libcgroup-0.41-8.el7.x86_64                                                                                                            2/2 

Installed:
  docker-engine.x86_64 0:1.8.0-1.el7.centos                                                                                                               

Dependency Installed:
  libcgroup.x86_64 0:0.41-8.el7                                                                                                                           

Complete!

[root@cent7 ~]# systemctl status docker
docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; disabled)
   Active: inactive (dead)
     Docs: https://docs.docker.com
[root@cent7 ~]# systemctl start docker
Job for docker.service failed. See 'systemctl status docker.service' and 'journalctl -xn' for details.
[root@cent7 ~]# journalctl -xn
-- Logs begin at Sun 2015-07-12 09:27:56 EDT, end at Sun 2015-07-12 09:34:04 EDT. --
Jul 12 09:32:34 cent7 systemd[1]: Starting Docker Socket for the API.
-- Subject: Unit docker.socket has begun with start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit docker.socket has begun starting up.
Jul 12 09:32:34 cent7 systemd[1]: Listening on Docker Socket for the API.
-- Subject: Unit docker.socket has finished start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit docker.socket has finished starting up.
-- 
-- The start-up result is done.
Jul 12 09:32:34 cent7 systemd[1]: Starting Docker Application Container Engine...
-- Subject: Unit docker.service has begun with start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit docker.service has begun starting up.
Jul 12 09:32:34 cent7 kernel: loop: module loaded
Jul 12 09:32:40 cent7 docker[2603]: time="2015-07-12T09:32:40.871943813-04:00" level=info msg="Option DefaultDriver: bridge"
Jul 12 09:32:40 cent7 docker[2603]: time="2015-07-12T09:32:40.872733230-04:00" level=info msg="Option DefaultNetwork: bridge"
Jul 12 09:32:40 cent7 docker[2603]: time="2015-07-12T09:32:40.896696252-04:00" level=warning msg="Running modprobe bridge nf_nat br_netfilter failed with messag
Jul 12 09:34:04 cent7 systemd[1]: docker.service operation timed out. Terminating.
Jul 12 09:34:04 cent7 systemd[1]: Failed to start Docker Application Container Engine.
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit docker.service has failed.
-- 
-- The result is failed.
Jul 12 09:34:04 cent7 systemd[1]: Unit docker.service entered failed state.
```

Expected to see docker package install correctly by default and allow using systemd to startup docker on the machine.

This is running inside a fresh install of Centos 7.1 inside virtualbox.  Docker 1.7.1 works with no issue.

### PR DESCRIPTION
**Pull request by jessfraz**
fix #15498

This creates our own docker-selinux package for use with our repo

ping @ewindisch @diogomonica @NathanMcCauley @vbatts @rhatdan

### CODE REVIEW

**Comment by ewindisch** [comment]
:-1:  I would like to have clarity on the license. If this is based on https://github.com/fedora-cloud/docker-selinux which is dual-licensed under the LGPL (LICENSE file) and GPLv2 (see docker-selinux.spec), then there could be a barrier to including this policy. Neither license is compatible with the Apache license: https://www.apache.org/legal/resolved.html#category-x

@jfrazelle could you verify if this is the source you used, or if there is some other package you've derived from? I would also appreciate if @rhatdan might have input. Perhaps there is a package (new or old) shipping with a policy we're able to bundle?

Worst-case we might need to ship as a separate package?

**Comment by jfrazelle** [comment]
I used a weird tarball from a fedora git url, hang on I believe it said GNU

**Comment by jfrazelle** [comment]
This is the tarball http://pkgs.fedoraproject.org/repo/pkgs/docker/docker-selinux-6267b83.tar.gz/226d2fc08263e698fb46cb52e414bea1/docker-selinux-6267b83.tar.gz

**Comment by jfrazelle** [comment]
So the way I pacakged the rpm it is a seperate package that the
docker-engine would then rely on... IANAL but it looks like red hat also
ships the seperate package as GPL and then Docker as Apache... but ya I
hvae noo idea

**Comment by ewindisch** [comment]
Checking more carefully at the commit, because it creates a new RPM it's probably fine.

**Comment by rhatdan** [comment]
As long as it is a separate RPM I believe it is fine.  IANAL.  You are
not actually using code inside of docker, these are completely separate.

**Comment by Djelibeybi** [comment]
I mentioned this to @jfrazelle on IRC, but `selinux-policy-targeted` on Fedora (20, 21, 22), CentOS 7 and Oracle Linux 7 already contains a `docker.pp` which allows Docker 1.8.1 to work. 

Perhaps instead of including another SElinux policy, it's worth reviewing the upstream policy and just adding a `Requires:` to `selinux-policy-targeted` for the `docker-engine` RPM?

**Comment by jfrazelle** [comment]
We want to control our own policy to ensure things don't break.

File: hack/make/.build-rpm/docker-engine.spec [hunk 1]
```diff
@@ -50,6 +50,27 @@ Requires: kernel-uek >= 3.8
...
+%global with_selinux 1
+%endif
+
+# start if with_selinux
+%if 0%{?with_selinux}
+# Version of SELinux we were using
+%if 0%{?fedora} >= 22
+%global selinux_policyver 3.13.1-119
+%else
+%global selinux_policyver 3.13.1-39
```
Comment (Djelibeybi) [inline hack/make/.build-rpm/docker-engine.spec]: This is too high for EL7-derived distros. The latest RPM in EL7 is `selinux-policy-3.13.1-23.0.1.el7_1.13.noarch`

Comment (jfrazelle) [inline hack/make/.build-rpm/docker-engine.spec]: I will fix for oracle

Comment (jfrazelle) [inline hack/make/.build-rpm/docker-engine.spec]: or 7 sorry

Comment (Djelibeybi) [inline hack/make/.build-rpm/docker-engine.spec]: Yeah, it's the same RPM version for CentOS7 (and RHEL7).

Comment (jfrazelle) [inline hack/make/.build-rpm/docker-engine.spec]: updated

**Comment by Djelibeybi** [comment]
I'm getting the following error trying to build `docker-engine` in `docker-engine.spec`:
    error: line 59: Unclosed %if

**Comment by jfrazelle** [comment]
fixed moved too fast

**Comment by rhatdan** [comment]
We have been asked by the upstream developers of selinux-policy (Which I used to be primary) to start shipping our own policy.  Getting fixes into RHEL/Centos selinux-policy base is too slow for a fast moving project like docker.  It only gets updated every 6-9 months.  The docker-selinux policy replaces and updates the docker.pp package.

**Comment by Djelibeybi** [comment]
Thanks @rhatdan, makes perfect sense.

**Comment by ewindisch** [comment]
LGTM

**Comment by Djelibeybi** [comment]
I'm testing this now, but the new spec file logic still looks weird to me. It's also 7am and I haven't had enough coffee, so I'm running a full RPM build to check before I comment further though.

**Comment by jfrazelle** [comment]
I built them all and tested






On Fri, Aug 28, 2015 at 2:29 PM -0700, "Avi Miller" <notifications@github.com> wrote:












I'm testing this now, but the new spec file logic still looks weird to me. It's also 7am and I haven't had enough coffee, so I'm running a full RPM build to check before I comment further though.

**Comment by jfrazelle** [comment]
But let me know what you find






On Fri, Aug 28, 2015 at 2:29 PM -0700, "Avi Miller" <notifications@github.com> wrote:












I'm testing this now, but the new spec file logic still looks weird to me. It's also 7am and I haven't had enough coffee, so I'm running a full RPM build to check before I comment further though.

**Comment by Djelibeybi** [comment]
Yep, already the first CentOS 7 build shows that the `Requires:` to `selinux-policy` is using the wrong version. I'm going to let it all finish, then will comment completely.

**Comment by jfrazelle** [comment]
Is that on oracle? Because I tested cents 7? I'm OOO today but can fix it later






On Fri, Aug 28, 2015 at 2:46 PM -0700, "Avi Miller" <notifications@github.com> wrote:












Yep, already the first CentOS 7 build shows that the Requires: to selinux-policy is using the wrong version. I'm going to let it all finish, then will comment completely.

**Comment by jfrazelle** [comment]
Ugh maybe I didn't push the latest /o\






On Fri, Aug 28, 2015 at 2:50 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Is that on oracle? Because I tested cents 7? I'm OOO today but can fix it later






On Fri, Aug 28, 2015 at 2:46 PM -0700, "Avi Miller" <notifications@github.com> wrote:












Yep, already the first CentOS 7 build shows that the Requires: to selinux-policy is using the wrong version. I'm going to let it all finish, then will comment completely.

**Comment by Djelibeybi** [comment]
That could be it. :) The logic appears to be all back-to-front on this latest commit.

**Comment by Djelibeybi** [comment]
(Copying this here from the code comment so that @jfrazelle is pinged via email too).

I know @jfrazelle said that this may not be the latest push, but this logic is pretty wrong. It sets `3.13.1-23` for Fedora 22 (which is actually the `Requires:` for Fedora 21) and then sets `3.13.1-39` for (almost) everything else, because non-Fedora distros will return `0` for `0%{?fedora}`

Somehow Fedora 21 doesn't get a `Requires:` for `selinux-policy` at all.

**Comment by jfrazelle** [comment]
Unfortunately I'm out of town at a wedding so this may have to wait til Monday sorry :/






On Fri, Aug 28, 2015 at 5:13 PM -0700, "Avi Miller" <notifications@github.com> wrote:












(Copying this here from the code comment so that @jfrazelle is pinged via email too).



I know @jfrazelle said that this may not be the latest push, but this logic is pretty wrong. It sets 3.13.1-23 for Fedora 22 (which is actually the Requires: for Fedora 21) and then sets 3.13.1-39 for (almost) everything else, because non-Fedora distros will return 0 for 0%{?fedora}



Somehow Fedora 21 doesn't get a Requires: for selinux-policy at all.

**Comment by jfrazelle** [comment]
Also fedora 21 is EOL






On Fri, Aug 28, 2015 at 5:15 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Unfortunately I'm out of town at a wedding so this may have to wait til Monday sorry :/






On Fri, Aug 28, 2015 at 5:13 PM -0700, "Avi Miller" <notifications@github.com> wrote:












(Copying this here from the code comment so that @jfrazelle is pinged via email too).



I know @jfrazelle said that this may not be the latest push, but this logic is pretty wrong. It sets 3.13.1-23 for Fedora 22 (which is actually the Requires: for Fedora 21) and then sets 3.13.1-39 for (almost) everything else, because non-Fedora distros will return 0 for 0%{?fedora}



Somehow Fedora 21 doesn't get a Requires: for selinux-policy at all.

**Comment by jfrazelle** [comment]
So if you are building from our scripts that dockerfile was removed it should not generate it at all






On Fri, Aug 28, 2015 at 5:16 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Also fedora 21 is EOL






On Fri, Aug 28, 2015 at 5:15 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Unfortunately I'm out of town at a wedding so this may have to wait til Monday sorry :/






On Fri, Aug 28, 2015 at 5:13 PM -0700, "Avi Miller" <notifications@github.com> wrote:












(Copying this here from the code comment so that @jfrazelle is pinged via email too).



I know @jfrazelle said that this may not be the latest push, but this logic is pretty wrong. It sets 3.13.1-23 for Fedora 22 (which is actually the Requires: for Fedora 21) and then sets 3.13.1-39 for (almost) everything else, because non-Fedora distros will return 0 for 0%{?fedora}



Somehow Fedora 21 doesn't get a Requires: for selinux-policy at all.

**Comment by jfrazelle** [comment]
Nevermind was thinking of fedora 20, regardless I'll take a look later






On Fri, Aug 28, 2015 at 5:17 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










So if you are building from our scripts that dockerfile was removed it should not generate it at all






On Fri, Aug 28, 2015 at 5:16 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Also fedora 21 is EOL






On Fri, Aug 28, 2015 at 5:15 PM -0700, "Jessica Frazelle" <jess@docker.com> wrote:










Unfortunately I'm out of town at a wedding so this may have to wait til Monday sorry :/






On Fri, Aug 28, 2015 at 5:13 PM -0700, "Avi Miller" <notifications@github.com> wrote:












(Copying this here from the code comment so that @jfrazelle is pinged via email too).



I know @jfrazelle said that this may not be the latest push, but this logic is pretty wrong. It sets 3.13.1-23 for Fedora 22 (which is actually the Requires: for Fedora 21) and then sets 3.13.1-39 for (almost) everything else, because non-Fedora distros will return 0 for 0%{?fedora}



Somehow Fedora 21 doesn't get a Requires: for selinux-policy at all.

**Comment by Djelibeybi** [comment]
No worries. Enjoy the weekend and the wedding. :)

**Comment by thaJeztah** [comment]
Thanks for testing and reviewing, @Djelibeybi, much appreciated to have some extra eyes on this :+1:

**Comment by jfrazelle** [comment]
Just rebased and force pushed what i had locally, so mad at myself for that sorry, would have gone the whole weekend mad about it haha

also thanks for testing, let me know if you find anything w these

**Comment by Djelibeybi** [comment]
Heh, makes total sense to me (which is why I'm reviewing on Saturday morning here in Melbourne). My reading of the updated commit looks OK, so I've triggered a full build of all RPMs just to be 100% sure.

Also, good excuse to test the performance of Oralce Linux in VMware Fusion 8 Pro. :)

**Comment by hqhq** [comment]
Tested on RHEL7, I can confirm it fixes the problem, thanks.

**Comment by Djelibeybi** [comment]
Built and tested on Fedora 21, 22, CentOS 7 and Oracle Linux: confirmed as fixed.

:+ LGTM.

**Comment by jfrazelle** [comment]
\o/ ping @diogomonica @nathanmccauley @ewindisch @icecrime merge it 0:)

**Comment by icecrime** [comment]
LGTM!

**Comment by thaJeztah** [comment]
thanks @jfrazelle and @Djelibeybi. Great work!

**Comment by Djelibeybi** [comment]
@jfrazelle and @thaJeztah: how do we mark this for cherry-picking into the 1.8 stream, out of interest?

**Comment by jfrazelle** [comment]
We won't do that I don't think, it's too large a change I say we wait to roll out for 1.9.0






On Sat, Aug 29, 2015 at 2:45 PM -0700, "Avi Miller" <notifications@github.com> wrote:












@jfrazelle and @thaJeztah: how do we mark this for cherry-picking into the 1.8 stream, out of interest?

**Comment by Djelibeybi** [comment]
So we're not going to have SElinux support for the EL7 distros for the life of 1.8?

**Comment by jfrazelle** [comment]
Yes there is SELinux support you can download the policy distributed by red hat etc. 






On Sat, Aug 29, 2015 at 2:59 PM -0700, "Avi Miller" <notifications@github.com> wrote:












So we're not going to have SElinux support for the EL7 distros for the life of 1.8?

**Comment by jfrazelle** [comment]
If you read the issue that is mentioned here that this fixes there is a very easy way to install that ;) so I don't think we should cherry pick a large change






On Sat, Aug 29, 2015 at 2:59 PM -0700, "Avi Miller" <notifications@github.com> wrote:












So we're not going to have SElinux support for the EL7 distros for the life of 1.8?

**Comment by Djelibeybi** [comment]
Gotcha. Will let our documentation team to document the requirement to ensure that selinux-policy-targeted is installed if SElinux is going to be used.

**Comment by moxiegirl** [comment]
@thaJeztah added changelog to this seems significant.  Remove if I overstepped of course.

**Comment by thaJeztah** [comment]
Thanks @moxiegirl, SGTM. Also, any maintainer is allowed to add those labels, so no worries!

**Pull request merged**